[Java] Updating the codelab text based on new template and styles

florina-muntenescu opened the pull request.

ceruleanotter commented:
Changes look good  - I compared the code that I got when starting from empty activity in 3.6 and following steps, here were some additional diffs which I think we might as well fix if we have time.

All of the code shown below is code from the codelab copy-pasta text (not this repo)

**WordListAdapter** (null checks for mWords):

 ```
   @Override
    public void onBindViewHolder(WordViewHolder holder, int position) {
        if (mWords != null) {
            Word current = mWords.get(position);
            holder.wordItemView.setText(current.getWord());
        } else {
            // Covers the case of data not being ready yet.
            holder.wordItemView.setText("No Word");
        }
    }
```

```
    @Override
    public int getItemCount() {
        if (mWords != null)
            return mWords.size();
        else return 0;
    }
```

**WordDao** (getAllWords different name and insert word doesn't have conflict strat)
```
    @Insert
    void insert(Word word);

    @Query("DELETE FROM word_table")
    void deleteAll();

    @Query("SELECT * from word_table ORDER BY word ASC")
    LiveData<List<Word>> getAllWords();
```

**WordRoomDatabase** - no mention of `.fallbackToDestructiveMigration()`

**WordViewModel** `insert` is public

florina-muntenescu commented:
> Changes look good - I compared the code that I got when starting from empty activity in 3.6 and following steps, here were some additional diffs which I think we might as well fix if we have time.
> 
> All of the code shown below is code from the codelab copy-pasta text (not this repo)
> 
> **WordListAdapter** (null checks for mWords):
Removed the initialisation of `mWords` and used the code in the codelab
> 
> ```
>   @Override
>    public void onBindViewHolder(WordViewHolder holder, int position) {
>        if (mWords != null) {
>            Word current = mWords.get(position);
>            holder.wordItemView.setText(current.getWord());
>        } else {
>            // Covers the case of data not being ready yet.
>            holder.wordItemView.setText("No Word");
>        }
>    }
> ```
> 
> ```
>     @Override
>     public int getItemCount() {
>         if (mWords != null)
>             return mWords.size();
>         else return 0;
>     }
> ```
> 
> **WordDao** (getAllWords different name and insert word doesn't have conflict strat)
Renamed to `getAlphabetizedWords`. Added the conflict strategy in the codelab text.
> 
> ```
>     @Insert
>     void insert(Word word);
> 
>     @Query("DELETE FROM word_table")
>     void deleteAll();
> 
>     @Query("SELECT * from word_table ORDER BY word ASC")
>     LiveData<List<Word>> getAllWords();
> ```
> 
> **WordRoomDatabase** - no mention of `.fallbackToDestructiveMigration()`
removed from code
> 
> **WordViewModel** `insert` is public
it's public in the codelab text and in the code.
